src/test/java/HomeWork.java

```diff
@@ -0,0 +1,80 @@
+        nameField.sendKeys("New Smart");
+        //Step8: Click on the Save button using xpath //div[@class="smart-playlist-form"]//footer/button[@type="submit"]
+        WebElement saveButton = wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//div[@class=\"smart-playlist-form\"]//footer/button[@type=\"submit\"]")));
+        saveButton.click();
+        //Step9: Confirm Success message using notification xpath: //div[@class="alertify-logs top right"]/div[@class='success show']
+        WebElement successMessage = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[@class=\"alertify-logs top right\"]/div[@class='success show']")));
+        //Step10: messageExpected is Updated playlist "New Smart."
+        Assert.assertEquals(successMessage.getText(),"Updated playlist \"New Smart.\"" );
+    }
+    @Test
+    public void renamePlaylist(){
+        //Step1: login to the website.
+        enterEmail("nayana.rao.subramanya@testpro.io");
+        enterPassword("Zqmvyk4hDaZ3vga");
+        loginButton();
+        Actions action = new Actions(driver);
+        //Step2: Navigate to the chosen playlist using xpath
+        WebElement chosenPlaylist = driver.findElement(By.xpath("//section[@id='playlists']/ul//a[@href='#!/playlist/87052']"));
+        wait.until(ExpectedConditions.visibilityOf(chosenPlaylist)).click();
+        //Step3: Right-click on the chosen playlist using actions
+        action.moveToElement(chosenPlaylist).doubleClick(chosenPlaylist).build().perform();
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("input[type='text']")));
+        //Step4: Choose the edit option using locator
+        /*WebElement editButton = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//section[@id='playlists']/ul/li[5]//li[1]")));
+        wait.until(ExpectedConditions.elementToBeClickable(editButton)).click();
+        Thread.sleep(1000);*/
+        //Step5: Clear the field
+        WebElement playlistNewNameField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//section[@id='playlists']/ul/li[@class='playlist playlist editing']/input[@type='text']")));
+        action.moveToElement(chosenPlaylist).doubleClick(playlistNewNameField).sendKeys(Keys.BACK_SPACE);
+        playlistNewNameField.sendKeys(Keys.chord(Keys.CONTROL,"a"),Keys.BACK_SPACE);
+        //Step6: Enter the new name
+        playlistNewNameField.sendKeys("New name");
+        //Step7: Press the enter key
+        playlistNewNameField.sendKeys(Keys.RETURN);
+        //Step8: Confirm we received the notification "Updated playlist \"New Playlist.\""
+        String successMessage = "Updated playlist \"New name.\"";
+        WebElement messageOfSuccess = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[@class=\"alertify-logs top right\"]/div[@class='success show']")));
+        Assert.assertEquals(messageOfSuccess.getText(),successMessage);
+
+    }
+
+}
```

Comment:
If possible, avoid overly long methods. A long method is 20‒25 lines. If possible, you should split the method into several shorter ones or even create a new class for this.

For example

   **Use in test methods, the following logical split:   GIVEN – WHEN - THEN**

    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```